synapse/module_api/__init__.py

```diff
@@ -1340,6 +1342,75 @@ async def store_remote_3pid_association(
         """
         await self._store.add_user_bound_threepid(user_id, medium, address, id_server)
 
+    def check_push_rule_actions(
+        self, actions: List[Union[str, Dict[str, str]]]
+    ) -> bool:
+        """Checks if the given push rule actions are valid according to the Matrix
+        specification.
+
+        See https://spec.matrix.org/v1.2/client-server-api/#actions for the list of valid
+        actions.
+
+        Added in Synapse v1.58.0.
+
+        Args:
+            actions: the actions to check.
+
+        Returns:
+            True if the actions are valid, False otherwise.
+        """
+        try:
+            check_actions(actions)
+        except InvalidRuleException:
+            return False
```

Comment:
given we're exposing `InvalidRuleException` anyway, why not let the exception propagate? it will give the caller more info than a simple boolean.